src/components/mediaControls.js

```diff
@@ -130,6 +127,30 @@ class MiniMediaPlayerMediaControls extends LitElement {
     }
   }
 
+  renderPlayButtons(hide) {
+    if (!hide.play_pause) {
+      return html`
+                <paper-icon-button
+                  @click=${e => this.player.playPause(e)}
+                  .icon=${ICON.PLAY[this.player.isPlaying]}>
+                </paper-icon-button>
+                ${!hide.play_stop ? html`
+                <paper-icon-button
+                  @click=${e => this.player.stop(e)}
+                  .icon=${ICON.STOP}>
+                </paper-icon-button>
+                ` : html``}
+                `;
+    } else if (!hide.play_stop) {
+      return html`
+                <paper-icon-button
+                  @click=${e => this.player.playStop(e)}
+                  .icon=${this.player.isPlaying ? ICON.STOP : ICON.PLAY[this.player.isPlaying]}>
+                </paper-icon-button>
+                `;
+    }
```

Comment:
This feels unnecessary, we basically have duplicate code for the `play_stop` button here.
No need for the if else here, just return a html template with each button wrapped inside an `${!hide... ? ... : html``}` statement

Reply:
It is not duplicated code. The click action and the icon are different when play_pause is hidden or not.
In the else, the stop button acts like a play/stop, in the oposite in the IF the stop is just stop button.

Reply:
I get that,
I would propose something like this:
```js
renderPlayButtons(hide) {
  return html`
    ${!hide.play_pause ? html`
      <paper-icon-button
        @click=${e => this.player.playPause(e)}
        .icon=${ICON.PLAY[this.player.isPlaying]}>
      </paper-icon-button>
    ` : html``}
    ${!hide.play_stop ? html`
      <paper-icon-button
        @click=${e => this.handleStop(e)}
        .icon=${hide.play_pause ? ICON.STOP[this.player.isPlaying] : ICON.STOP.true}>
      </paper-icon-button>
    ` : html``}
  `;
}

handleStop(e) {
  return this.config.hide.play_pause ? this.player.playStop(e) : this.player.stop(e);
}
```

Wouöd requirre a true, false for `ICON.STOP` const, similar to `ICON.PLAY`

Reply:
Sorry for the delay.

I don't like the fact that handleStop method depends directly on hide option. Hide should be relative to rendering stuff. We could move the code from handleStop to @click handler. What do you think?